### DISCUSSION TITLE
提取pag帧画面时，定义类型为pag::ColorType::BGRA_8888但是保存下来的图片好像没有透明

### DISCUSSION THREAD

**Discussion by jackTJT** [Tencent/libpag#1913] [General] [answered]
使用以下代码保持图片的帧画面，发现保存下来的图片没有透明通道
		pag_player->setProgress(currentFrame * 1.0 / totalFrames);
		auto status = pag_player->flush();

		I_LOG("---currentFrame:[{}], flushStatus:[{}]", currentFrame, status);

		//auto data = new uint8_t[bytesLength];
		cv::Mat image;
		image.create(pagFile->height(), pagFile->width(), CV_8UC4);
		pagSurface->readPixels(pag::ColorType::BGRA_8888, pag::AlphaType::Premultiplied, image.data,
			image.step);
以下是放到ppt中的效果图
<img width="588" alt="a408fd687e452c73d943b66826ee1c0d" src="https://github.com/Tencent/libpag/assets/96832322/7cfae1aa-faaa-4904-aa00-98e429dfc4d6">
想问一下是哪里出了什么问题，或者是pag提取的数据alpha默认为255吗

**Comment by domchen** [comment, marked as answer]
是这个PAG 素材本身就带了白色不透明背景图，换个有透名背景的素材截图。
